R/plot_sankey.R

```diff
@@ -0,0 +1,85 @@
+#' Create a sankey plot
+#'
+#' @param data A data frame like [toy_data].
+#'
+#' @return
+#' @export
+#'
+#' @examples
+#' plot_sankey(toy_data)
+plot_sankey <- function(data) {
+  data_links <- data |>
+    mutate(
+      source = "bank",
+      target = "PCTR_risk_category",
+      value = "amount",
+      middle_node2 = "tilt_sec"
+    )
+
+  links <- data_links |>
+    select(
+      source = "bank",
+      target = "tilt_sec",
+      value = "amount",
+      group = "PCTR_risk_category"
+    )
+  if ("middle_node2" %in% names(data_links)) {
```

Comment:
FIXME: I think the `if` branch always run (and the `else` branch never runs) because the code above adds the column `middle_node2` to `data_links`.